src/components/Billing/Invoice/EditInvoiceTable.tsx

```diff
@@ -94,18 +167,46 @@ export function EditInvoiceTable({
 }: EditInvoiceTableProps) {
   const { t } = useTranslation();
   useShortcutSubContext("facility:billing:invoice:show");
+
+  // Helper function to create a unique key for discount component
+  const getDiscountComponentKey = (
+    component: MonetaryComponent | undefined,
+  ) => {
+    if (!component?.code?.code) return undefined;
+    return component.code.code;
+  };
+
+  // Helper function to get available discount components for a charge item
+  const getAvailableDiscounts = (chargeItem: ChargeItemRead) => {
+    return (
+      chargeItem?.charge_item_definition?.price_components?.filter(
+        (component) =>
+          component.monetary_component_type === MonetaryComponentType.discount,
+      ) || []
+    );
+  };
+
+  const getTaxComponents = (chargeItem: ChargeItemRead) => {
+    return (
+      chargeItem?.charge_item_definition?.price_components?.filter(
+        (component) =>
+          component.monetary_component_type === MonetaryComponentType.tax,
+      ) || []
+    );
+  };
+
```

Comment:
Can we create a single function to extract the required component from ChargeItem?


```suggestion
  const getPriceComponents = (chargeItem: ChargeItemRead, tyep: MonetaryComponentType) => {
    return (
      chargeItem?.charge_item_definition?.price_components?.filter(
        (component) =>
          component.monetary_component_type === type,
      ) || []
    );
  };

```

Reply:
Done